qtff/src/file.rs

```diff
@@ -65,6 +68,42 @@ impl File {
         }
     }
 
+    fn get_moof_data(&mut self) -> Result<Vec<MovieFragment>> {
+        let mut fragments = vec![];
+        loop {
+            let start_position = self.f.stream_position()?;
+
+            let atom = match AtomReader::new(&mut self.f).find(|a| match a {
+                Ok(a) => a.typ == MovieFragment::TYPE,
+                Err(_) => true,
+            }) {
+                Some(Ok(a)) => a,
+                Some(Err(err)) => return Err(err.into()),
+                None => return Ok(fragments),
+            };
+            let mut buf = Vec::new();
+            atom.data(&mut self.f).read_to_end(&mut buf)?;
+
+            let mut movie_fragment = MovieFragment::read(Cursor::new(buf.as_slice()))?;
+            movie_fragment.fragment_header.start_position = start_position;
+            fragments.push(movie_fragment);
+        }
+    }
+
+    pub fn get_movie_fragment_data(&mut self) -> Result<Vec<MovieFragment>> {
```

Comment:
Naming suggestion since this returns a collection of `MovieFragment`s:

```suggestion
    pub fn get_movie_fragments(&mut self) -> Result<Vec<MovieFragment>> {
```